filtering scripts/filtering.py

```diff
@@ -0,0 +1,50 @@
+from pathlib import Path
+from mdframe.reader import run, Config
+from filtering_helper import dataframe_check
+
+def filter_files(nutrition_subgroup : str = None, food_type : str = None, description : str = None, weight : float = None, 
```

Comment:
This method of filtering overfits to the format of the file. Try to think of a more general-purpose way of filtering, where you e.g. take a string name of an attribute from the file and check it against a value. This should be a relatively simple wrapper around what Pandas already does with `df.filter` or Pandas indexing, e.g. `df[df[attrname] == value]`.

Reply:
Tip: `**kwargs` might be helpful here

Reply:
Despite the fact that we use a particular format for the metadata TOML we would like to be able to apply filtering to _any_ dataset. For example, what's going to happen if you have a file like:

```toml
some_other_id = 123
some_other_attribute = "hello"
```

What if I wanted to filter by `some_other_attribute`?